pkg/logger/logger.go

```diff
@@ -0,0 +1,81 @@
+package logger
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+)
+
+const (
+	logLevelDebug = 2
+	logLevelWarn  = 3
+	logLevelTrace = 4
+)
+
+// Wrapper defines a convenient interface to use to log things.
+type Wrapper interface {
+	Info(msg string, keysAndValues ...any)
+	Debug(msg string, keysAndValues ...any)
+	Warn(msg string, keysAndValues ...any)
+	Trace(msg string, keysAndValues ...any)
+	Error(err error, msg string, keysAndValues ...any)
+	WithValues(keysAndValues ...any) *Logger
+	WithName(name string) *Logger
+	GetLogger() logr.Logger
```

Comment:
This is called GetLogger because simply calling it `Logger` would have clashed with the name and caused a bunch of weird problems. :)